### IMPORTAZIONE LIBRERIE

In [79]:

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.cluster import KMeans
from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
from sklearn.linear_model import LinearRegression, LogisticRegression
from sklearn.metrics import accuracy_score, mean_squared_error, silhouette_score
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder, StandardScaler

print ("librerie")
print("prova")

librerie
prova


#### Definizione stile grafico: sfondo bianco,griglia leggera,grafici moderni

In [80]:
plt.style.use('seaborn-v0_8-whitegrid')
%matplotlib inline

#### Read dataset fumatori italia

In [81]:
df = pd.read_csv('fumatori_italia.csv')

### OSSERVAZIONE HEAD E TAIL
#### Abbiamo osservato la parte iniziale e finale del dataset per vedere come era composto

In [82]:
df.head()

,Anno,Sesso,Nord_ovest,Nord_est,Centro,Sud,Isole,Italia
0,1980,Maschi,52.3,48.6,53.0,61.5,56.5,54.3
1,1983,Maschi,45.3,41.0,44.5,50.6,46.1,45.6
2,1987,Maschi,42.3,34.7,39.3,43.8,43.3,40.8
3,1990,Maschi,37.6,31.7,37.5,41.7,40.9,37.8
4,1993,Maschi,34.6,29.4,34.8,38.8,38.6,35.1


In [83]:
df.tail()

,Anno,Sesso,Nord_ovest,Nord_est,Centro,Sud,Isole,Italia
47,2011,Femmine,18.0,16.2,19.2,14.3,14.1,16.6
48,2012,Femmine,17.9,16.5,17.8,13.5,16.3,16.4
49,2013,Femmine,16.1,15.6,18.6,13.2,15.5,15.8
50,2014,Femmine,15.5,14.9,15.9,12.7,14.8,14.7
51,2015,Femmine,15.4,15.4,17.3,12.6,14.4,15.0


In [84]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 52 entries, 0 to 51
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Anno        52 non-null     int64  
 1   Sesso       52 non-null     str    
 2   Nord_ovest  52 non-null     float64
 3   Nord_est    52 non-null     float64
 4   Centro      52 non-null     float64
 5   Sud         52 non-null     float64
 6   Isole       52 non-null     float64
 7   Italia      52 non-null     float64
dtypes: float64(6), int64(1), str(1)
memory usage: 3.4 KB


In [85]:
df.describe()

,Anno,Nord_ovest,Nord_est,Centro,Sud,Isole,Italia
count,52.000000,52.000000,52.000000,52.000000,52.000000,52.000000,52.000000
mean,2001.076923,24.800000,23.053846,25.826923,24.505769,24.794231,24.609615
std,9.480312,8.541571,6.759089,8.024768,12.101376,10.905070,9.128693
min,1980.000000,15.400000,14.900000,15.900000,11.900000,11.700000,14.700000
25%,1995.000000,17.875000,18.075000,19.375000,13.950000,15.175000,16.975000
50%,2001.500000,21.650000,21.400000,22.750000,21.300000,21.250000,21.250000
75%,2009.000000,30.150000,27.400000,31.675000,33.025000,32.850000,31.050000
max,2015.000000,52.300000,48.600000,53.000000,61.500000,56.500000,54.300000


### MINIMO E MASSIMO:
###### Il valore massimo in Italia arriva a 54.3.
###### Il valore massimo più alto si trova nel Sud, con 61.5.
###### Questo indica che nei primi anni del dataset la percentuale di fumatori era molto alta.

In [86]:
# GRAFICO MIN E MAX

### MEDIA:
###### In molte colonne la media è più alta della mediana.
###### Questo significa che i valori alti degli anni iniziali alzano la media.


In [87]:
#Grafico media

### MEDIA E DEVIAZIONE STANDARD:
###### Sud e Isole hanno una deviazione standard più alta.Questo indica maggiore variazione nel tempo.


#grafico media e std


### ANALISI PERCENTILI:
###### 25° Percentile: il 25% delle osservazioni totali si attesta sotto questo valore.
###### 50° Percentile(mediana): divide esattamente a metà il dataset. Essendo i dati maschili e femminili in numero uguale.
###### In molte colonne la media è leggermente più alta del 50%.
###### Questo indica che alcuni anni con percentuali elevate aumentano la media generale.
###### 75° Percentile: il 75% dei dati è inferiore a questo valore,
###### I valori sopra il 75% rappresentano gli anni con percentuali di fumatori più elevate.
### Differenza tra 25% e 75%:
#### La distanza tra questi percentili mostra la dispersione dei dati nel tempo.
##### Italia:
###### mostra una diminuzione progressiva dei fumatori nel corso degli anni.

In [88]:
#grafico analisi 